### PR TITLE
feat: add Swagger API Gateway integration

### LINKED ISSUE #116: Create Swagger Spec Resource in SAM Template
The SAM template.yaml file now needs to explicitly define a new resource type called *AWS::Serverless::Api*

https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessapi

### PR DESCRIPTION
**Pull request by jasonwalsh**
This pull request integrates [Swagger](https://swagger.io/) with API Gateway using [API Gateway Extensions](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions.html).

> How does it work?

The `swagger.yaml` file contains a definition of the current HTTP API which includes the following endpoints:

| Endpoint |
|----------|
| `/square/locations` |
| `/square/redirect` |
| `/square/oauth` |

Using the [`x-amazon-apigateway-integration`](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-integration.html) object, API Gateway is able to model the HTTP API after the Swagger specification.

> What are the benefits?

There are a few benefits. For starters, Swagger is a commonly used specification for documenting and designing HTTP APIs. Swagger has the ability to generate [documentation](https://app.swaggerhub.com/apis-docs/hopdrops/hopdrops/1.0.0) and integrates with third-party services, such as API Gateway.

Swagger also allows for documenting required data, such as headers, request bodies, and query string parameters for endpoints. API Gateway is able to validate the request before invoking the integration function. If the request is invalid (e.g., a query string parameter is missing), then API Gateway rejects the request.

The following screenshot shows an example of the `/square/locations` **Method Request** configuration:

Screen Shot 2019-09-25 at 1 56 34 PM

This means that any request received from the client *must* include `code` as a query string parameter.

This pull request also includes a custom API Gateway [Authorizer](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-use-lambda-authorizer.html).

> What is an API Gateway Authorizer?

From AWS:

> A Lambda authorizer (formerly known as a custom authorizer) is an API Gateway feature that uses a Lambda function to control access to your API.

There are two types of authorizers:

- A token-based Lambda authorizer (also called a TOKEN authorizer) receives the caller's identity in a bearer token, such as a JSON Web Token (JWT) or an OAuth token
- A request parameter-based Lambda authorizer (also called a REQUEST authorizer) receives the caller's identity in a combination of headers, query string parameters, stageVariables, and $context variables

This pull request uses a `TOKEN` authorizer.

All of the code used in the `requires_auth` decorator has been reused in the authorizer function.

> Does this change how I make requests to the HTTP API?

No. The HTTP API remains the same. This integration is purely internal and does not affect downstream users.

Resolves #116 #117 #118

### CODE REVIEW

File: functions/locations.py [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
-try:
-    from decorators import requires_auth
-except ImportError:
-    from .decorators import requires_auth
 
-
-@requires_auth
 def handler(event, context):
-    client = Client(access_token=event['headers']['X-Access-Token'])
+    logger.info(event)
```
Comment (brh55) [inline functions/locations.py]: Do we want to log the event given the access token will be exposed in the logs?

Comment (jasonwalsh) [reply functions/locations.py]: Good point. I had this here for debugging purposes. I'll remove it.

Comment (jasonwalsh) [reply functions/locations.py]: Removed as of f3ca0e2

**Pull request merged**